### PR TITLE
Allow formatted markup for node descriptions

### PR DESCRIPTION
**Pull request by mPokornyETM**
See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

### Proposed changelog entries

* Allow HTML syntax for node descriptions.
Before
image

After
image

### Proposed upgrade guidelines

No

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
      There are no changes in java code.
- [x] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
      There are no changes in java code.
- [x] For dependency updates: links to external changelogs and, if possible, full diffs
      There are no changes in java code.

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6511"></a>

### CODE REVIEW

**Review by NotMyFault — COMMENTED** [review]
This is a nice approach, but I'd recommend to stick to what's already in place.
For description input we're using a codemirror box, like seen on job or user descriptions, with optional syntax support, if enabled in the security realm.

Elevating a jenkins-input to be able to render HTML breaks with the existing handling for these kinds of text input.

**Comment by mPokornyETM** [comment]
> 

I try to find some example. There is editableDescrption page but it does not work on this page. Because this page has an extra config page. Has you some idea?

**Comment by NotMyFault** [comment]
> > 
> 
> I try to find some example. There is editableDescrption page but it does not work on this page. Because this page has an extra config page. Has you some idea?

Changing the input from a textbox to a textarea in hudson.slaves.DumbSlave.configure-entries.jelly pointing the markup formatter to codemirror does generate named box.
Existing spots are for example hudson.model.Job.configure.jelly utilizing this feature.

**Comment by mPokornyETM** [comment]
So editable config description works too.

image
-->
image

The button to 'swicht temporally offline' is moved too. So it has the same look like in configure job page.
@NotMyFault  thanks for hits.

**Comment by daniel-beck** [comment]
How does this look in matrix-project? I think that renders node descriptions and should be adapted.

Did you check whether project history explains why nodes have plain text `nodeDescription` but everything else has formatted `description`?

**Comment by mPokornyETM** [comment]
> How does this look in matrix-project? I think that renders node descriptions and should be adapted.
No I have no idea, newer use it, but I will try it and get feedback here

> Did you check whether project history explains why nodes have plain text `nodeDescription` but everything else has formatted `description`?
Nope, I was just angry, because we can not use longer description. We want to fill the description dynamically to get some node specific info when is started / used. Ex:
+ installed tools and their versions like git, java ...
+ real platform name (the platform labeler plugin does not work as expected for us)
+ IpConfig
+ Reboot history, we need to reboot the nodes many times during our 'build' process
....

So we need bigger description field.

But I will try to find the reason. It was also terrible for me to fix it, because you true anywhere else it is `description`

**Comment by mPokornyETM** [comment]
> matrix-project

So the matrix-project renders node labels. Node name and description are not changed.
image
image
I think my changes has no effect on this.

**Comment by mPokornyETM** [comment]
> Did you check whether project history explains why nodes have plain text `nodeDescription` but everything else has formatted `description`?

So I have try to find the reason, It looks that it was just "for ever". When you agree, I would create new Jira issue for that. So we can "fix" it little later. I prefer to set the 'nodeDescription' to obsolete, but this can not be decided by me.

**Comment by mPokornyETM** [comment]
I found one more usage of nodeDescription
image

I has also check whole jenkins core ws for nodeDescription and I think this was the last one.

**Comment by daniel-beck** [comment]
@mPokornyETM Thanks! To clarify, I'm not opposed to this change per se, just wondering whether there's a good reason for the discrepancy in both field name and behavior. I expect we'll merge this or a variant in the end, solving your problem.

Re matrix-project, I was surprised by your answer and checked, and while
> Screenshot 2022-04-27 at 13 33 58
looks like there would be an agent description rendered there, there's no support for that -- everything is treated as a label, not as an agent name, internally: https://github.com/jenkinsci/matrix-project-plugin/blob/7aea491852f39b51379a2f6f2dd490f7d191441e/src/main/java/hudson/matrix/LabelAxis.java#L90-L99
Additionally, the label descriptions do not get the renderer applied, so it looks terrible with any non-default markup formatter. That's an issue with the plugin alone though.

**Comment by mPokornyETM** [comment]
@daniel-beck  Ah sorry that was misunderstanding from my side. I think it renders some hove the `/computer/<agentName>` page, but it is used here in the config page from the matrix-project self.
The solution is strange. From my side I prefer to remove the description from this page, because it is not necessary and it looks like the `lol ()` is a function and not the node `name (description)`

Please allow me one question. What shall be done to finish this PR?

**Comment by daniel-beck** [comment]
> Please allow me one question. What shall be done to finish this PR?

At the moment, needs more reviews from others, and consequently some patience from you. If anything is missing from this PR, that'll be brought up in reviews.

**Comment by daniel-beck** [comment]
Nit:

The description of the built-in node is defined by `Hudson.NodeDescription` in `hudson/model/Messages.properties` as

> the Jenkins controller's built-in node

That probably looks weird with this change in presentation. It should be upgraded to a full sentence since it's no longer a short phrase in parentheses (no HTML though, or any characters likely to lead to problem in any common(ish) markup language).

**Comment by mPokornyETM** [comment]
> > Please allow me one question. What shall be done to finish this PR?
> 
> At the moment, needs more reviews from others, and consequently some patience from you. If anything is missing from this PR, that'll be brought up in reviews.

thanks, I am patient enough, but I am excited, Like a small child

**Comment by mPokornyETM** [comment]
dirty fix for padding
image

**Comment by daniel-beck** [comment]
> > It should be upgraded to a full sentence since it's no longer a short phrase in parentheses (no HTML though, or any characters likely to lead to problem in any common(ish) markup language).
> 
> Hm I see. The same issue was in `System Information` page. My quick and dirty solution was `<div style="margin-bottom: var(--section-padding);">`

To clarify, the comment you quoted was about the phrasing, not the presentation.

Thanks for taking care of the margin!

**Comment by mPokornyETM** [comment]
Hi @daniel-beck . Any news here?

File: core/src/main/resources/hudson/model/Computer/index.jelly
Comment (alecharp) [inline core/src/main/resources/hudson/model/Computer/index.jelly:37]: I think there is a problem here because the icon and the text are not aligned vertically. I'm trying to find a solution for this.

**Review by alecharp — CHANGES_REQUESTED** [review]
I'm here requesting for change because I think there is a mis-alignement with the text and icon. **But**, I think this is a good improvement which only need a small adjustment.

File: core/src/main/resources/hudson/model/Computer/index.jelly
Comment (alecharp) [reply core/src/main/resources/hudson/model/Computer/index.jelly:37]: I removed the `<span/>` and added the `icon-*` to the `<l:icon/>` but the class is not passed to the svg (and the icon is 512px by 512px). Adding the class to the resulting SVG is however fixing the icon size.

Comment (alecharp) [reply core/src/main/resources/hudson/model/Computer/index.jelly:37]: we might have a larger issue with the `symbol` because the class is not passed and I could only fix this with editing the `icon` layout and adding the `icon-md` to https://github.com/jenkinsci/jenkins/blob/0f0a5097ed0a45b78d58e976dd29d32e9f6a62ac/core/src/main/resources/lib/layout/icon.jelly#L71 because the `iconMetadata.classSpec` is always empty.

@janfaracik could you help here? maybe I'm over-thinking the problem here.

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:37]: thx any help are welcome

Comment (janfaracik) [reply core/src/main/resources/hudson/model/Computer/index.jelly:37]: > we might have a larger issue with the `symbol` because the class is not passed and I could only fix this with editing the `icon` layout and adding the `icon-md` to
> 
> https://github.com/jenkinsci/jenkins/blob/0f0a5097ed0a45b78d58e976dd29d32e9f6a62ac/core/src/main/resources/lib/layout/icon.jelly#L71
> 
> because the `iconMetadata.classSpec` is always empty.
> @janfaracik could you help here? maybe I'm over-thinking the problem here.

Hey, I spotted that too (https://github.com/jenkinsci/design-library-plugin/pull/67#issuecomment-1134005433) - changing it to this 



will fix the missing class issue, I've given a quick look around Jenkins and it doesn't seem to have broken anything but I can't guarantee that.

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:37]: @janfaracik  So when I understand it correctly, there are no change request for this PR. This shall be done somewhere else (global) right?

Comment (alecharp) [reply core/src/main/resources/hudson/model/Computer/index.jelly:37]: @mPokornyETM I still have a problem. Because you added a `<span/>` with the `icon-lg` class, the icon is not aligned correctly. The code should be something like `<h1><l:icon/>Title</h1>` imo.

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:36]: I don`t know, hwy it shows as change, But I haven´t change the part
```
<span class="icon-lg">
  <l:icon src="${it.iconClassName}" />
</span>
```

**Review by alecharp — APPROVED** [review]
This looks fine. We need to improve the code once the icon attributes is fixed (@janfaracik https://github.com/jenkinsci/jenkins/pull/6511/files#r879816769). 

Once this is fixed, we can revisit the `<span class="icon-lg"/>` as the icon is not `lg` anyway but `md`.

**Comment by janfaracik** [comment]
> This looks fine. We need to improve the code once the icon attributes is fixed (@janfaracik https://github.com/jenkinsci/jenkins/pull/6511/files#r879816769).
> 
> Once this is fixed, we can revisit the `<span class="icon-lg"/>` as the icon is not `lg` anyway but `md`.

Happy to say the fix for the symbol attributes is merged in :) https://github.com/jenkinsci/jenkins/pull/6609

**Comment by alecharp** [comment]
> > This looks fine. We need to improve the code once the icon attributes is fixed (@janfaracik https://github.com/jenkinsci/jenkins/pull/6511/files#r879816769).
> > Once this is fixed, we can revisit the `<span class="icon-lg"/>` as the icon is not `lg` anyway but `md`.
> 
> Happy to say the fix for the symbol attributes is merged in :) #6609

I totally missed that. So please @mPokornyETM, you could please review your commit to leverage that fix?

**Review by alecharp — CHANGES_REQUESTED** [review]
Waiting to fix the icon part of the page.

**Comment by mPokornyETM** [comment]
Hi of cores I do that.
Compiling ... waiting ... ;-) but this shall be not a big deal

File: core/src/main/resources/hudson/model/Computer/index.jelly
Comment (alecharp) [inline core/src/main/resources/hudson/model/Computer/index.jelly:35]:
```suggestion
        <l:icon src="${it.iconClassName}" class="icon-xlg" />
```

**Review by alecharp — COMMENTED** [review]
@mPokornyETM sorry, I should have been more specific: with #6609, you can use class attribute directly in the `l:icon` now.

File: core/src/main/resources/hudson/model/Computer/index.jelly
Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:35]: I has try the solution too, but I think there was no space between icon and text. I will retry it little later, because I am compiling other branch now. And are you sure 'xlg'? That means extra-large, right?

Comment (alecharp) [reply core/src/main/resources/hudson/model/Computer/index.jelly:35]: here is what I have locally with my suggestion

Screen Shot 2022-06-10 at 18 03 09

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:35]: @alecharp  So I do quick tests
```
<h1><l:icon src="${it.iconClassName}" class="icon-xlg" />${it.caption}</h1>
<h1><l:icon src="${it.iconClassName}" class="icon-lg" />${it.caption}</h1>
<h1><l:icon src="${it.iconClassName}" class="icon-md" />${it.caption}</h1>
<h1><l:icon src="${it.iconClassName}" class="icon-sm" />${it.caption}</h1>
```
 and it looks like this
image

 Which one you find the best? I do not like any of them (do not ask why? just it looks terrible for me ;-))

**Review by NotMyFault — APPROVED** [review]

File: core/src/main/resources/hudson/model/Computer/index.jelly
Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly]: thx. I see. Nice article. You true, I do not like it too. It is more or less trash. But in this case it is necessary, because it represent node online / offline state. Otherwise the whole page shall be reworked as well. An I want only change the description. (general I do not understand, why no body change the pages about nodes, labels, node_minotoring ... . It is
one of the most important part in Jenkins, but I will do my best to make the pages more modern and fit with jenkins design-library fit - of cores in other PR :-)) again thx for review.

Comment (timja) [reply core/src/main/resources/hudson/model/Computer/index.jelly]: Yes absolutely not suggesting a change here :)

**Comment by daniel-beck** [comment]
Two remarks testing the latest build:

1. The built-in node's description still looks weird now that it's on a standalone line.
2. Inline editing of the description is not possible. Was this previously brought up and dismissed already? If not, is there a reason this wasn't added?

**Comment by mPokornyETM** [comment]
> Two remarks testing the latest build:
> 
> 1. The built-in node's description still looks weird now that it's on a standalone line.
Strange.  I will check it.
> 2. Inline editing of the description is not possible. Was this previously brought up and dismissed already? If not, is there a reason this wasn't added?
This was also my first try, but there is extra configuration page, where you can edit the description. So I revert it. But I think it is low effort to allow both variants. Shall I implement it?

**Comment by daniel-beck** [comment]
>  Strange. I will check it.

To clarify, it's the specific string from Messages.properties that e.g. starts with a lowercase letter. That made sense in parentheses, less so with the new presentation.

>  This was also my first try, but there is extra configuration page, where you can edit the description. So I revert it. But I think it is low effort to allow both variants. Shall I implement it?

I think that would be useful to have, especially as it would be consistent with every other "description" field in Jenkins (Views, Jobs, Builds, Users, all have inline description editing)

**Comment by mPokornyETM** [comment]
So the description in built-in node shall looks like before
But the inline description does not work for me :-(
I have added 
```
  <t:editableDescription permission="${it.CONFIGURE}" />
```
and after save I get following esception:
```
404 Not Found
Stapler processed this HTTP request as follows, but couldn't find the resource to consume the request

-> evaluate(<hudson.model.Hudson@1d0ce1eb> :hudson.model.Hudson,"/computer/firstAgent/submitDescription")
-> evaluate(((StaplerProxy)<hudson.model.Hudson@1d0ce1eb>).getTarget(),"/computer/firstAgent/submitDescription")
-> evaluate(<hudson.model.Hudson@1d0ce1eb>.getComputer(),"/firstAgent/submitDescription")
-> evaluate(<hudson.model.ComputerSet@59a6b974> :hudson.model.ComputerSet,"/firstAgent/submitDescription")
-> evaluate(<hudson.model.ComputerSet@59a6b974>.getDynamic("firstAgent",...),"/submitDescription")
-> evaluate(<hudson.slaves.SlaveComputer@6c08d0f3> :hudson.slaves.SlaveComputer,"/submitDescription")
-> evaluate(((StaplerProxy)<hudson.slaves.SlaveComputer@6c08d0f3>).getTarget(),"/submitDescription")
-> evaluate(<hudson.slaves.SlaveComputer@6c08d0f3>.getDynamic("submitDescription",...),"")
            hudson.slaves.SlaveComputer@6c08d0f3.getDynamic("submitDescription",...)==null. Back tracking.
-> No matching rule was found on <hudson.slaves.SlaveComputer@6c08d0f3> for "/submitDescription"
```

Sorry I have no idea how to fix it.

**Comment by daniel-beck** [comment]
Yeah, that's because there's no `#doSubmitDescription` method on the `Computer` you're editing. You'll need to add that, and process the inline submission there.

**Comment by daniel-beck** [comment]
> So the description in built-in node shall looks like before

No, it looks good with the change. It's just the exact wording no longer works when it's a full standalone paragraph. Just replace

> the Jenkins controller's built-in node

with

> This is the Jenkins controller's built-in node.

or if you want to make it even nicer,

> This is the Jenkins controller's built-in node. Builds running on this node will execute on the same system and as the same user as the Jenkins controller. This is appropriate e.g. for special jobs performing backups, but in general you should run builds on agents.

(Or something along those lines.)

**Comment by daniel-beck** [comment]
Is https://github.com/daniel-beck/jenkins/commit/ae3e1c0ccaf1b7a07d4d3e039d3d22ff0b8e5456 something you could see as part of your PR? WDYT?

**Comment by daniel-beck** [comment]
Now I remember why I gave up implementing this a few years ago. There's so much crap hardcoded in inline description editing. I'll give it a try…

**Comment by daniel-beck** [comment]
https://github.com/daniel-beck/jenkins/commit/1e866ce1935a4526bd7aebd3f3a7e39065345611 implements inline editing for agent descriptions for your consideration. Please note the TODOs that will need to be addressed before it would be merged.

**Comment by mPokornyETM** [comment]
@daniel-beck thx too much. I am next 2 weeks on vacation (without laptop ...) So I will check it after them.

**Comment by mPokornyETM** [comment]
> Is [daniel-beck@ae3e1c0](https://github.com/daniel-beck/jenkins/commit/ae3e1c0ccaf1b7a07d4d3e039d3d22ff0b8e5456) something you could see as part of your PR? WDYT?

Very good idea. It will be more readable and decrease maintenance. Can you merge your changes in to my branch please. Thx

One thing. It will be better when the page is named common-info.jelly because now it can be used also in other pages (Where is the node info showed - like in /label/<someNodeLabel>

File: core/src/main/java/hudson/slaves/SlaveComputer.java
Comment (daniel-beck) [inline core/src/main/java/hudson/slaves/SlaveComputer.java:795]: It is unclear to me what these messages mean.

File: core/src/main/resources/hudson/slaves/SlaveComputer/systemInfo.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/slaves/SlaveComputer/systemInfo.jelly:44]: Why this change? (other than extracting the nodeDescription into its own element)

File: core/src/main/resources/lib/hudson/editable-description.js
Comment (daniel-beck) [inline core/src/main/resources/lib/hudson/editable-description.js:2]: This works if and only if the `#description-link` element is already present when the page is loaded.

I think that's always the case, but would welcome counter-examples.

File: core/src/main/resources/hudson/model/Computer/index-top.jelly
Comment (daniel-beck) [inline core/src/main/resources/hudson/model/Computer/index-top.jelly:3]: Is this an unrelated change?

File: core/src/main/resources/hudson/slaves/SlaveComputer/systemInfo.jelly
Comment (mPokornyETM) [reply core/src/main/resources/hudson/slaves/SlaveComputer/systemInfo.jelly:44]: Because the node description is also used in this page. But here shall be the description not changed. From my side, I will remove the description here. It is more or less useless information on this page.

File: core/src/main/java/hudson/slaves/SlaveComputer.java
Comment (mPokornyETM) [reply core/src/main/java/hudson/slaves/SlaveComputer.java:795]: It might happens, that you edit the description, but the node is deleted between (some other user or you in other session, or some groovy script or ... ). Any better English translations are welcome.

File: core/src/main/resources/lib/hudson/editable-description.js
Comment (mPokornyETM) [reply core/src/main/resources/lib/hudson/editable-description.js:2]: This was done by you.

File: core/src/main/resources/hudson/model/Computer/index-top.jelly
Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index-top.jelly:3]: You true, it makes no more sense, because there are used t:editableDescription. Of cores aI will remove the line.

File: core/src/main/resources/hudson/slaves/SlaveComputer/systemInfo.jelly
Comment (daniel-beck) [reply core/src/main/resources/hudson/slaves/SlaveComputer/systemInfo.jelly:44]: To clarify, changing from h1 to h1 + h2.

File: core/src/main/resources/lib/hudson/editable-description.js
Comment (daniel-beck) [reply core/src/main/resources/lib/hudson/editable-description.js:2]: Correct. This comment is not negative feedback, but providing context for other reviewers.

File: core/src/main/resources/hudson/model/Computer/index-top.jelly
Comment (daniel-beck) [reply core/src/main/resources/hudson/model/Computer/index-top.jelly:3]: I think simple-page is used to support the login and user creation pages that do not use the usual layout with header and sidepanel, so it is wrong here.

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index-top.jelly:3]: Good, that we see it equal. So I am not sure how shall be solve this kind of style issues:
1. Inline style -- dirty solution
2. use a some global css file - you will lost the overview, where is the style used.
3. create new css files (in this case 3 times) with the same content makes also not really sense.


So can we 'vote' which solution will be better. I vote for 1. Dirty, but ease. I think the "style="margin-bottom: var(--section-padding);" is not so much overhead in the files.

File: core/src/main/resources/hudson/slaves/SlaveComputer/systemInfo.jelly
Comment (mPokornyETM) [reply core/src/main/resources/hudson/slaves/SlaveComputer/systemInfo.jelly:44]: I will try to build my jenkins tomorrow again. and paste here screenshots before && after. 
The page System Information looks very strange when you use some description.
Maybe we shall really remove the part with the description from this site. It is useless here. What you think?

File: core/src/main/resources/hudson/model/Computer/index-top.jelly
Comment (timja) [reply core/src/main/resources/hudson/model/Computer/index-top.jelly:3]: If you're just after some spacing see https://weekly.ci.jenkins.io/design-library/Spacing/

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index-top.jelly:3]: @timja  grat, this might solve my issue . thx

File: core/src/main/resources/hudson/slaves/SlaveComputer/systemInfo.jelly
Comment (daniel-beck) [reply core/src/main/resources/hudson/slaves/SlaveComputer/systemInfo.jelly:44]: >  Maybe we shall really remove the part with the description from this site. It is useless here. What you think?

Seems like a good idea, it doesn't work like that with anything else.

File: core/src/main/java/hudson/slaves/SlaveComputer.java
Comment (daniel-beck) [reply core/src/main/java/hudson/slaves/SlaveComputer.java:795]: I will propose improvements 👍

Comment (mPokornyETM) [reply core/src/main/java/hudson/slaves/SlaveComputer.java]: Make it easy. I think it is more sure now.

**Comment by mPokornyETM** [comment]
@daniel-beck  thx for merge

**Comment by basil** [comment]
I merged the physical conflicts but now there is a logical conflict:

```
core/src/main/resources/lib/hudson/editable-description.js
   9:16  error  'replaceDescription' is not defined  no-undef
  11:16  error  'replaceDescription' is not defined  no-undef
```

**Comment by mPokornyETM** [comment]
@daniel-beck  thx for caring.

File: core/src/main/resources/jenkins/model/Jenkins/MasterComputer/index-top.properties
Comment (daniel-beck) [inline core/src/main/resources/jenkins/model/Jenkins/MasterComputer/index-top.properties:4]:
```suggestion
  <a rel="noopener noreferrer nofollow" target="_blank" href="http://www.jenkins.io/redirect/distributed-builds/">Learn more about distributed builds.</a>
```

**Review by daniel-beck — COMMENTED** [review]
@mPokornyETM Sorry for the delay here!

TBH I'm trying to figure out what is blocking this. I got assigned after providing early feedback. I provided a suggestion addressing early problems, which you integrated. That had open TODOs that needed resolving, but those have been resolved and the TODOs removed.

Is it just that I didn't unassign myself, so nobody else went near this? I'll try to see whether changing that gets others to take a look 😃 (as a substantial co-author I'm unsure whether I should be approving this.)

**Review by timja — APPROVED** [review]
LGTM

----

There's one UI issue but likely pre-existing issue with the component / other buttons on the page.

The Save and mark this node temporarily offline buttons don't look that great as save is wider than the form and they are touching

image

File: core/src/main/java/hudson/slaves/SlaveComputer.java
Comment (NotMyFault) [inline core/src/main/java/hudson/slaves/SlaveComputer.java:785]:
```suggestion
    /**
     * Inline editing of description
     * @since TODO
     */
```

**Review by NotMyFault — APPROVED** [review]

File: core/src/main/java/hudson/slaves/SlaveComputer.java
Comment (timja) [reply core/src/main/java/hudson/slaves/SlaveComputer.java:785]: It's not part of the public Java API so it shouldn't have `@since` could be restricted though

Comment (NotMyFault) [reply core/src/main/java/hudson/slaves/SlaveComputer.java:785]: If it's not intended to be public API, we should restrict it then 👍🏻

**Comment by timja** [comment]
/label ready-for-merge

----

This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

File: core/src/main/resources/hudson/model/Computer/index.jelly
Comment (NotMyFault) [inline core/src/main/resources/hudson/model/Computer/index.jelly:64]: Causes [JENKINS-70394](https://issues.jenkins.io/browse/JENKINS-70394)

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:64]: I does not see any reason, why? Shall I fixed it, or work somebody on that?

Comment (daniel-beck) [reply core/src/main/resources/hudson/model/Computer/index.jelly:64]: Indirectly; it now is to the right of a `div.warning` with `position: relative` whose area overlaps this button and doesn't allow interacting with the button.

No clue how to resolve other than move elements around which wouldn't be a good fix.

Comment (timja) [reply core/src/main/resources/hudson/model/Computer/index.jelly:64]: heh it'll be related to https://github.com/jenkinsci/jenkins/pull/6511#pullrequestreview-1233350846 with the pre-existing UI issue. This sort of thing has happened before, there's a box around one of the elements I think which can block interacting with others.

> Shall I fixed it, or work somebody on that?

Would be great if you could

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:64]: We will see, I will try it today (or better tonight ;-) )

Comment (daniel-beck) [reply core/src/main/resources/hudson/model/Computer/index.jelly:64]: > as save is wider than the form

It's unclear to me what the problem identified in the linked comment is, as the Save button being to the right of the description field is everywhere in Jenkins:


Screenshot 2023-01-10 at 17 53 08

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:64]: After few months pause in Jenkins core I need little more time to setup my dev instance again. But anyway , I tries to fix this buttons. It breaks the view in all my combination. Therefore I tries something "new". The app-bar as describe here: https://weekly.ci.jenkins.io/design-library/AppBar/

In has implemented app-bar and bottomButtonBar as well to see how it can looks. Therefore you see the button twice.
image
It works perfect. The question is : will be this applicable solution? And when which one is preferred. I personally like the bottomButtonBar, because it can be placed close to reason. But I my look-and-feel is not the best one (I am working more on backends in my daily business)


This is how it looks on "small" screen
image


and when you scroll up, you newer lost focus to this buttons:
image

Anyway, please get me feedback here which one (or no one) solution will be fine. Then I will commit my changes and create a PR.

ps: sorry for wasting here, but I will spare time preparing PR and discussing there if it is fine or not. I thing this will be faster way, because many of your core guys are mentioned here ;-)

Comment (timja) [reply core/src/main/resources/hudson/model/Computer/index.jelly:64]: I think app-bar would make the most sense. I don't see why those buttons would need to be sticky

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:64]: @timja you right, I play now with the panel (try to add node monitors as table in the same view) and definitely is better app-bar. That means PR coming soon.

Comment (mPokornyETM) [reply core/src/main/resources/hudson/model/Computer/index.jelly:64]: see PR #7577

**Pull request merged**
